packages/ButtonGroup/src/button-group.scss

```diff
@@ -5,233 +5,242 @@
-
-  &.is-icon-only {
-    width: var(--ids-button-group-item-icon-only-width);
-  }
-
-  &:active,
-  &:hover,
-  &:focus,
-  &.is-selected {
-    z-index: var(--ids-button-group-item-index-max);
-  }
-
-  &:first-child {
-    border-radius: var(--ids-button-group-item-border-radius-left);
-  }
-
-  &:last-child {
-    border-radius: var(--ids-button-group-item-border-radius-right);
-  }
-
-  &:hover {
-    background-color: var(--ids-button-group-item-background-hover);
-    border-color: var(--ids-button-group-item-border-color-hover);
-    color: var(--ids-button-group-item-text-hover);
-    cursor: pointer;
-  }
-
-  &:focus {
-    outline: 0;
-    background: var(--ids-button-group-item-background-hover);
-    box-shadow: var(--ids-button-group-item-focus);
-  }
-
-  &:not(:first-child) {
-    margin-left: var(--ids-button-group-item-margin);
-  }
-
-  &.is-selected {
-    border-color: var(--ids-button-group-item-border-selected);
-    color: var(--ids-button-group-item-text-selected);
-    background-color: var(--ids-button-group-item-background-selected);
-    z-index: var(--ids-button-group-item-index-selected);
-  }
-
-  &.is-disabled {
-    border-color: var(--ids-button-group-item-border-disabled);
-    background-color: var(--ids-button-group-item-background-disabled);
-    color: var(--ids-button-group-item-text-disabled);
-    pointer-events: none;
-    user-select: none;
-  }
+    //The following declarations are required to support the tooltip in the ButtonGroup
+    > *:first-child .ids-btn-group__item {
+        border-radius: var(--ids-button-group-item-border-radius-left);
+    }
+
+    > *:last-child .ids-btn-group__item {
+        border-radius: var(--ids-button-group-item-border-radius-right);
+    }
+
+    > *:not(:first-child) {
+        margin-left: var(--ids-button-group-item-margin);
+    }
 }
 
-.ids-btn-group--compact {
-  .ids-btn-group__item {
-    padding-left: var(--ids-button-group-item-padding-compact);
-    padding-right: var(--ids-button-group-item-padding-compact);
+.ids-btn-group__item {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    font-family: inherit;
+    font-size: inherit;
+    font-weight: inherit;
+    position: relative;
+    border-style: solid;
+    border-color: var(--ids-button-group-item-border-color);
+    border-width: var(--ids-button-group-item-border-width);
+    border-radius: var(--ids-button-group-item-border-radius-middle);
+    background-color: var(--ids-button-group-item-background);
+    margin: 0;
+    padding: 0 var(--ids-button-group-item-padding);
+    z-index: var(--ids-button-group-item-index);
 
     &.is-icon-only {
-      width: var(--ids-button-group-item-icon-only-width-compact);
+        width: var(--ids-button-group-item-icon-only-width);
+    }
+
+    &:active,
+    &:hover,
+    &:focus,
+    &.is-selected {
+        z-index: var(--ids-button-group-item-index-max);
+    }
+
+    &[data-item="is-first"] {
```

Comment:
I guess I don't really understand how this works.. Wouldn't this apply the border radius to the tooltips container if there is one? Or do we not care? How is this actually different from just using :first-child and :last-child?

Reply:
The use of `:first-child` and `:last-child` works very well as long as the notion of `Tooltip` is not introduced. However, when the Tooltip is added, the DOM changes, so these properties no longer fulfill their role correctly.

### Example with a tooltip
```
... 
<div class="ids-btn-group">
  <span class="ids-tooltip__container">
    <button class="ids-btn-group__item">
...
```
In this example, the class `ids-btn-group__item` is both `first-child` and `last-child`, and as the last declaration in the css file is `last-child`, it applies the border radius to the right for all children. 

To counter this problem, in the parent, we convert the children into an array and apply a data tag first to the first element of the array and last to the last. In this way, we can have a slightly more precise css rule.